src/components/IntlTelInputApp.js

```diff
@@ -152,41 +171,29 @@ class IntlTelInputApp extends Component {
     document.addEventListener('keydown', this.handleDocumentKeyDown);
   }
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.value !== nextProps.value) {
-      this.setState({
-        value: nextProps.value,
-      });
-    }
-
-    if (this.props.disabled !== nextProps.disabled) {
-      this.setState({
-        disabled: nextProps.disabled,
-      });
-    }
-
-    if (
-      typeof nextProps.customPlaceholder === 'function' &&
-      this.props.customPlaceholder !== nextProps.customPlaceholder
-    ) {
-      this.updatePlaceholder(nextProps);
-    }
-  }
-
-  componentWillUpdate(nextProps, nextState) {
+  shouldComponentUpdate(nextProps, nextState) {
     if (nextState.showDropdown) {
```

Comment:
Although it's work, but I'm not sure if it is an anti-pattern to put codes which is not related to component-update control. 🤔

Reply:
componentWillUpdate is gone. 

The only place left for component will update code to go is in `should component update`

Reply:
OK, got it. 👌